src/scripts/main.js

```diff
@@ -1,5 +1,33 @@
 'use strict';
 
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
 document.addEventListener('click', e => {
-  // write code here
+  const spiderCenter = spider.width / 2;
+
+  if (e.target !== wall) {
+    return;
+  }
+
+  let coordSpiderX = e.offsetX - spiderCenter;
+  let coordSpiderY = e.offsetY - spiderCenter;
+
+  if (e.offsetX < spiderCenter) {
+    coordSpiderX = 0;
+  }
+
+  if (e.offsetY < spiderCenter) {
+    coordSpiderY = 0;
+  }
+
+  if (e.offsetX > wall.clientWidth - spiderCenter) {
```

Comment:
in such cases, the sequence of actions is not obvious at first glance, so use parentheses